.github/workflows/try-runtime.yml

```diff
@@ -0,0 +1,64 @@
+name: Try-runtime
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened, edited]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+env:
+  TRY_RUNTIME_CLI_VERSION_TAG: v0.7.0
+  CARGO_HOME: ./.cargo
+
+jobs:
+  check_skip_flags:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Commit
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ env.GITHUB_SHA }}
+      - name: Get Head Commit Message
+        id: head-commit-message
+        run: echo "HEAD_COMMIT_MESSAGE=$(git show -s --format=%s)" >> "$GITHUB_OUTPUT"
+
+  test-try-runtime:
+    strategy:
+      matrix:
+        runtime: [peregrine, spiritnet]
+    runs-on: ubuntu-latest
+    needs: check_skip_flags
+    container:
+      image: paritytech/ci-unified:bullseye-1.74.0
+
+    if: ${{ !contains(needs.check_skip_flags.outputs.head-commit-message, 'ci-skip-rust') }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up caching for Cargo
+        uses: actions/cache@v4
+        with:
+          path: ${{ secrets.CARGO_CACHE_PATH }}
+          key: cargo-try-runtime-${{ matrix.runtime }}-${{ hashFiles('**/Cargo.lock') }}
+          restore-keys: |
+            cargo-try-runtime-${{ matrix.runtime }}-
+
+      - name: Run try-runtime
+        run: |
+          echo "Running ${{ matrix.runtime }} runtime migration check"
+
+          echo "---------- Downloading try-runtime CLI ----------"
+          curl -sL https://github.com/paritytech/try-runtime-cli/releases/download/${{ env.TRY_RUNTIME_CLI_VERSION_TAG }}/try-runtime-x86_64-unknown-linux-musl -o try-runtime
+          chmod +x ./try-runtime
+          echo "Using try-runtime-cli version:"
+          ./try-runtime --version
+
+          echo "---------- Building ${{ matrix.runtime }} runtime ----------"
+          cargo build --release --locked -p ${{ matrix.runtime }}-runtime --features try-runtime
+
+          echo "---------- Executing on-runtime-upgrade for ${{ matrix.runtime }} ----------"
+          ./try-runtime --runtime ./target/release/wbuild/${{ matrix.runtime }}-runtime/${{ matrix.runtime }}_runtime.compact.compressed.wasm \
+              on-runtime-upgrade --disable-spec-version-check --checks=all ${{ matrix.subcommand_extra_args }} live --uri wss://${{ matrix.runtime }}.kilt.io
```

Comment:
I guess `matrix.subcommand_extra_args` does not exist?

Reply:
There's an extension for GH Actions that would show these warnings: `github.vscode-github-actions`

Reply:
Replaced it with env variable